### ISSUE TITLE
[PERF] Regression on Json scenarios

### ISSUE THREAD

**Issue by sebastienros** [aspnet/KestrelHttpServer#2573]
Regression of approximately 10% on Json serialization scenarios between versions `2.2.0-preview1-34172` and `2.2.0-preview1-34173`, around May 11 12pm UTC.

Update: Apparently on Linux only

**Comment by sebastienros** [comment]
This commit: https://github.com/aspnet/KestrelHttpServer/commit/b040e33b1bbcc9719dc8e71c707a7e3fe1ff99ca

Source: http://aspnetci/viewLog.html?buildId=465847&buildTypeId=AspNetCore_Publish&tab=buildChangesDiv

**Comment by davidfowl** [comment]
Must have been the scheduling change

**Comment by davidfowl** [comment]
This should only have affected Linux

**Comment by sebastienros** [comment]
@davidfowl awesome, because this is exactly what I saw too and had already updated the first post

**Comment by DamianEdwards** [comment]
Helpe understand which versions are impacted here please? The linked commit was a merge from 2.1 to 2.2 (dev) so are we saying this is a regression in 2.1 RTM from what was previously in dev?

**Comment by benaadams** [comment]
> Must have been the scheduling change

`merged 2 commits into release/2.1 ... 4 days ago` https://github.com/aspnet/KestrelHttpServer/pull/2557 had the change

```diff
-            // On *nix platforms, Sockets already dispatches to the ThreadPool.
-            var awaiterScheduler = IsWindows ? _scheduler : PipeScheduler.Inline;
```

**Comment by davidfowl** [comment]
Yes, it's that one. 

/cc @halter73
